src/sentry_core.c

```diff
@@ -633,3 +656,94 @@ sentry_set_level(sentry_level_t level)
         scope->level = level;
     }
 }
+
+bool
+sentry__should_sample_transaction(sentry_value_t tx_cxt)
+{
+    sentry_value_t context_setting = sentry_value_get_by_key(tx_cxt, "sampled");
+    if (!sentry_value_is_null(context_setting)) {
+        return sentry_value_is_true(context_setting);
+    }
+
+    bool sample = false;
+    SENTRY_WITH_OPTIONS (options) {
+        sample = options->traces_sample_rate == 1.0;
+        // TODO: roll rng if sample rate >0 and <1.0
+        // TODO: run through traces sampler function if rate is unavailable
+    }
+    return sample;
+}
+
+sentry_value_t
+sentry_start_transaction(sentry_value_t tx_cxt)
+{
+    // TODO: stuff transaction into the scope
+    bool should_sample = sentry__should_sample_transaction(tx_cxt);
+
+    sentry_value_t tx = sentry_value_new_event();
+
+    // TODO: does it matter that this is sampled on creation vs on send? can we
+    // just sample on send like other events?
```

Comment:
good Q. I think since the sampling decision can also be set from the outside, it should be defined here?

Reply:
potentially adding spans could become a no-op if sampled on creation?  saves some cpu